cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -507,7 +507,7 @@ func (c *machineController) machineSetNodeGroups() ([]*nodegroup, error) {
 		if ng.MaxSize()-ng.MinSize() > 0 {
 			if ng.scalableResource.CanScaleFromZero() {
 				nodegroups = append(nodegroups, ng)
-			} else if pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, 0) > 0 {
+			} else if pointer.Int32PtrDerefOr(machineSet.Spec.Replicas, machineSet.Status.Replicas) > 0 {
```

Comment:
So, if a machineset cannot scale from zero (doesn't have proper annotations), Spec.Replicas might be nil while Status.Replicas is > 0.  This is because users might be applying a manifest that doesn't specify replicas after changing some other setting, and we ignore nil replicas currently.

This fix makes good sense to me.